### ISSUE TITLE
[Bug]: Radio keyboard focus ring shows after click

### ISSUE THREAD

**Issue by adixon-adobe** [adobe/spectrum-web-components#2998]
### Code of conduct

- [X] I agree to follow this project's code of conduct.

### Impacted component(s)

sp-radio

### Expected behavior

Radio keyboard focus ring only shows up when you use a keyboard interaction, similar to other components

### Actual behavior

If you go to https://opensource.adobe.com/spectrum-web-components/components/radio/ and click on a radio option, the keyboard focus ring displays along with the white selection styling.

### Screenshots

<img width="329" alt="Screen Shot 2023-03-08 at 1 04 42 PM" src="https://user-images.githubusercontent.com/48841507/223812275-ecb3a08e-a799-4687-8713-88dba2cf716b.png">

### What browsers are you seeing the problem in?

Chrome

**Comment by Westbrook** [comment]
Looks like this may have root in Spectrum CSS: https://unpkg.com/browse/@spectrum-css/radio@7.0.6/dist/index-vars.css#L157 @pfulton thoughts?

**Comment by pfulton** [comment]
> Looks like this may have root in Spectrum CSS: https://unpkg.com/browse/@spectrum-css/radio@7.0.6/dist/index-vars.css#L157 @pfulton thoughts?

I'm unable to reproduce this in the examples in our [docs](https://opensource.adobe.com/spectrum-css/radio.html). Is there something different about the structure of the component here vs. our markup?

**Comment by Westbrook** [comment]
Definitely some differences, the main question here is the use of `:focus` here as opposed to `.focus-ring` or `:focus-visible` as seen in other components.
